akka-docs/src/main/paradox/typed/reliable-delivery.md

```diff
@@ -0,0 +1,413 @@
+actors, resulting in fatal errors like `OutOfMemoryError` because too many messages are queued in the mailboxes
+of the actors. The detection of lost messages and the flow control is driven by the consumer side, which means
+that the producer side will not send faster than the demand requested by the consumer side. The producer side will
+not push resends unless requested by the consumer side.
+
+There are 3 supported patterns, which are described in the following sections:
+
+* @ref:[Point-to-point](#point-to-point)
+* @ref:[Work pulling](#work-pulling)
+* @ref:[Sharding](#sharding)
+
+## Point-to-point
+
+Point-to-point reliable delivery between a single producer actor sending messages and a single consumer actor
+receiving the messages.
+
+Messages are sent from the producer to @apidoc[ProducerController] and via @apidoc[ConsumerController] actors, which
+handle the delivery and confirmation of the processing in the destination consumer actor.
+
+![delivery-p2p-1.png](./images/delivery-p2p-1.png)
+
+The producer actor will start the flow by sending a `ProducerController.Start` message to
+the `ProducerController`.
+
+The `ProducerController` sends `RequestNext` to the producer, which is then allowed to send one
+message to the `ProducerController`. Thereafter the producer will receive a new `RequestNext`
+when it's allowed to send one more message.
+
+The producer and `ProducerController` actors are supposed to be local so that these messages are
+fast and not lost.
+
+Similarly, on the consumer side the destination consumer actor will start the flow by sending an
+initial `ConsumerController.Start` message to the `ConsumerController`. 
+
+For the `ProducerController` to know where to send the messages it must be connected with the
+`ConsumerController`. You do this is with `ProducerController.RegisterConsumer` or
+`ConsumerController.RegisterToProducerController` messages. When using the the point-to-point pattern
+it is the application's responsibility to connect them together. For example, by sending the `ActorRef`
+in an ordinary message to the other side, or register the `ActorRef` in the @ref:[Receptionist](actor-discovery.md)
+and find it on the other side.
+
+You must also take measures to reconnect them if any of the sides crashes, for example by watching it
+for termination.
+
+Received messages from the producer are wrapped in `ConsumerController.Delivery` when sent to the consumer,
+which is supposed to reply with `ConsumerController.Confirmed` when it has processed the message.
+Next message is not delivered until the previous is confirmed. More messages from the producer that arrive
+while waiting for the confirmation are stashed by the `ConsumerController` and delivered when the previous
+message is confirmed.
+
+The consumer and the `ConsumerController` actors are supposed to be local so that these messages are fast
```

Comment:
```suggestion
The consumer and the `ConsumerController` actors are guaranteed to be local so that these messages are fast
```

Reply:
That doesn't seem right, that's up to the user, it's a recommendation for the user and not an aspect built into the reliable delivery actors.

Reply:
it's enforced by a check. I'll reformulate this, but "guaranteed" is not the right word because user can do the wrong thing (but will then notice at runtime)

Reply:
clarified in https://github.com/akka/akka/pull/28155/commits/64178fa5a2ff41eb59a87c8d0178317ba5ae3dc1